server/test/deckTest.js

```diff
@@ -1,23 +1,104 @@
+
 chai.use(chaiHttp);
 
-// import chai from "chai";
-// import chaiHttp from "chai-http";
-// import server from "../index.js";
-// let should = chai.should();
-// chai.use(chaiHttp);
+// test expected values
+const testDeckName = "Spanish Vocabs";
+const testDeckId = "62b15d18e4693d1555a5e946";
+const invalidId = "123";
+const invalidIdErrorMessage = "Deck with ID " + invalidId + " not found";
+const invalidNameMessage = "Valid name required";
+const emptyNameErrorMessage = "Name cannot be empty";
+
+describe("POST Decks", function () {
+  it("POST, add one test deck and DELETE", function (done) {
+    chai
+      .request(server)
+      .post("/decks")
+      .send({ name: "test deck" })
+      .end(function (err, res) {
+        res.should.have.status(201);
+        res.should.be.json;
+        res.body.should.be.a("object");
+        res.body.should.have.property("name");
+        res.body.name.should.equal("test deck");
+        chai
+          .request(server)
+          .delete("/decks/" + res.body._id)
+          .end((err, res) => {
+            res.should.have.status(200);
+            res.body.should.be.a("object");
+            res.body.should.have.property("message").eql("Deck deleted");
+            done();
+          });
+      });
+  });
+
+  it("POST Error, test createDeck with no input", function (done) {
+    chai
+      .request(server)
+      .post("/decks")
+      .send({})
+      .end(function (err, res) {
+        res.should.have.status(409);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql("Name cannot be empty");
+        done();
+      });
+  });
+
+  it("POST Error, test createDeck with empty string name", function (done) {
+    chai
+      .request(server)
+      .post("/decks")
+      .send({ name: "" })
+      .end(function (err, res) {
+        res.should.have.status(409);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql(emptyNameErrorMessage);
+        done();
+      });
+  });
+
+  it("POST Error, test createDeck with whitespace string name", function (done) {
+    chai
+      .request(server)
+      .post("/decks")
+      .send({ name: " " })
+      .end(function (err, res) {
+        res.should.have.status(404);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql(invalidNameMessage);
+        done();
+      });
+  });
+
+  it("POST Error, test createDeck with special character string name", function (done) {
+    chai
+      .request(server)
+      .post("/decks")
+      .send({ name: "?!" })
+      .end(function (err, res) {
+        res.should.have.status(404);
+        res.body.should.be.a("object");
+        res.body.should.have.property("message").eql(invalidNameMessage);
+        done();
+      });
+  });
+});
 
-describe("Decks", function () {
-  it("list first deck GET", (done) => {
+describe("GET Decks", function () {
+  it("GET, get first deck in db", (done) => {
```

Comment:
I'm curious. Is there any way to retrieve all card decks when get is called? Or are we always return the first one?

Reply:
Yes but the issue is as mentioned in the edited comment above, number of decks in db will not be static. That's why for now I did the 1st deck in db as I've seen in the [example I shared](https://github.com/mjhea0/node-mocha-chai-tutorial/blob/master/test/test-server.js).

Reply:
My purpose of testing total number was to see if GET properly gets all the decks in db. To do this, it has to be static some way that's why I had to reset Flashcard collection in db in the beginning of testing in my case.. I know this method is not appropriate so I am trying to do it in other way. Your way of testing should be good then